mixins/imageHelper.ts

```diff
@@ -88,6 +262,8 @@ const imageHelper = Vue.extend({
      * @param {number} [options.maxHeight=element?.clientHeight] - Maximum height of the image in CSS pixels (optional).
      * @param {number} [options.quality=90] - Quality level of the image (optional, only relevant for jpeg format).
      * @param {boolean} [options.limitByWidth=false] - Use the element's width instead of its height for the size calculation.
+     * @param {number} [options.backdropIndex=0] - Index of the backdrop image to use (when image type is 'Backdrop').
+     * @param {boolean} [options.checkParent=true] - If it's needed to look for the parent's image tags in case the item's requested image type/tag doesn't exist.
      * @returns {string} The URL for the image, with the base URL set and the options provided.
      */
     getImageUrlForElement(
```

Comment:
Fallbacks should probably be handled by this. So pass it an item and an image type, and it'll handle the fallbacks.

There's mainly one that I can think of right now, which is `thumb -> backdrop`, but I believe web also falls back to Primary for that. 

This can technically be done later, but since this refactors pretty most of the function, it should probably take this into account from the get go to avoid further refactorings.

Reply:
@mrtimscampi My idea was that, in case we have many possibilities for the images, use the function directly in an v-if or computed property and check if it returns undefined or not.

Remember also that BlurhashImage has now the possibility to replace images that doesn't exist with placeholders, which can be even more useful.

Same for getBlurhash function, if we're not sure there's a blurhash, we should check first calling this function directly and checking for undefined

Handling all the possible fallbacks here would make the mixin really big for no really huge improvement. Majority of components can already call the mixin without no checks, but the scarce cases where we need it we only need two lines more of code.